### ISSUE TITLE
Way to view all currently applied ignored dependencies / versions

### ISSUE THREAD

**Issue by richardowen** [dependabot/dependabot-core#2255]
The ability to ignore dependencies or certain versions of dependencies is great if you can't upgrade to a certain version right now. However they then get lost.

It would be great if there was a way to see (perhaps in the Dependabot dashboard) the currently-ignored dependencies so they can be reviewed and ignore rules removed as required.

**Comment by infin8x** [comment]
Closing this as it's available in GitHub-native Dependabot.

**Comment by amacneil** [comment]
> it's available in GitHub-native Dependabot

@infin8x could you please share a link to docs or explain where to find this?

**Comment by infin8x** [comment]
I no longer work on the Dependabot team but @asciimike may be able to help

**Comment by planetf1** [comment]
^ As above - there seems no way to check what has been ignored currently.

**Comment by asciimike** [comment]
There are several ways to check what has been or is being ignored:
 - check your `dependabot.yml`'s `ignore` conditions
 - check the logs (as above)
 - search closed PRs for `@dependabot ignore`

I agree there isn't a holistic view in line with your dependencies--hopefully when we get to rebuilding that view, we can fit this in. Until then, the above is the recommended solution.

**Comment by planetf1** [comment]
@asciimike Thanks for the tip. The log was useful in particular. Having a way to manage in future would be good (including to remove the ignore). Great tool.

**Comment by benomatis** [comment]
> * check your `dependabot.yml`'s `ignore` conditions
Not an option if I don't want to keep ignoring a version

> * check the logs (as above)
Pretty much the only option, and not a great one, to say the least

> * search closed PRs for `@dependabot ignore`
Not an option as dependabot includes this text in ALL PR's, so it will return ALL PR's even merged ones.

**Comment by jeffwidman** [comment]
Re-opening, as this has come up several times both here in core and also internal support issues that it'd be helpful for users to have a self-service way to view all currently applied ignores filters.

Also related:
* https://github.com/dependabot/dependabot-core/issues/6489

**Comment by planetf1** [comment]
Related, would be how to 'unignore' a dependency

For example we've put off some updates for many months due to waiting on a larger spring6 update. For some that have been ignored we may spot some that we no longer want to ignore

**Comment by jeffwidman** [comment]
This shipped in:
* https://github.com/dependabot/dependabot-core/pull/7654

It's currently behind a feature flag, so won't go live for most users for a little bit yet, but we're getting close.

**Comment by honeyankit** [comment]
We have implemented a [feature](https://github.com/dependabot/dependabot-core/pull/7654) to show the ignore conditions on the PR description body. Currently it is behind feature flag and will be available soon (probably by next week) to all the users. Hence, closing this issue.

**Comment by xxchan** [comment]
@jeffwidman @honeyankit Is this feature available now? I can't see it in dependabot PRs. :eyes:

**Comment by jeffwidman** [comment]
I'm not sure, I'm no longer working on the core :dependabot: team.

@honeyankit did this ship?

In general, I prefer to not close an issue til it's fully available to users, ie an internal feature flag doesn't really qualify.

**Comment by zorzella** [comment]
The feature mentioned [above](https://github.com/dependabot/dependabot-core/issues/2255#issuecomment-1659031823) allows for showing the ignore conditions of one particular dep, but not all the deps in a repo.

The [documentation](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file) suggests `searching the repository for "@dependabot ignore" in:comments` but that brings any PR that was ever ignored and it's unwiedly.

I would still like to be able to see all the ignores in a project.

**Comment by torokati44** [comment]
+1 on the previous comment

**Comment by EthanZmuda** [comment]
I also agree that this doesn't seem to be resolved. The command implemented requires the user already knows the dependency may be ignored, or at least have the suspicion, and yes requires a PR.

**Comment by planetf1** [comment]
Agree with last few comments - still need a way to see the full list of ignores

**Comment by koniiiik** [comment]
Showing a list of ignored in a PR body doesn't solve this issue when dependabot doesn't create any PRs for a given dependency, once it has decided to ignore all available version. Is there any way we could get this implemented in the insights → dependency graph → dependabot UI? Adding a link to the ignore comment itself would be even nicer, as that would make it possible to easily look for the context behind the ignore condition.

**Comment by neilgierman** [comment]
I found this issue while trying to find the same information for my repositories. I did finally construct a gh cli + jq command that solves for me. I found that when you use `@dependabot ignore this dependency`, dependabot will add a comment with specific wording that is not used in other comments for other types of ignore commands. Posting here in case others need the same information before a better feature is released. Note, only tested on Linux.

```
gh pr list --author "app/dependabot" --state closed --limit 1000 --json number,title,comments | jq -r '.[] | select(.comments[].body | contains("again, unless you re-open this PR.")) | "\(.number): \(.title)"'
```

**Comment by wouterds** [comment]
Still that's very cumbersome, you might have hundreds if not thousands of PRs and (un)ignored certain dependencies multiple times.

**Comment by IamR3m** [comment]
What about command `@dependabot show ignore conditions` without `<dependency name>`?

**Comment by OJFord** [comment]
Does @honeyankit still work on Dependabot? This does not seem to have shipped, still, @jeffwidman - anyone else on Dependabot you might be able to point this way?

We could really do with some visibility on what's being ignored (possibly for reasons no longer relevant or true).

**Comment by torokati44** [comment]
_bump_

**Comment by cjmalloy** [comment]
My dependabot started listing ignores at the top of each new PR, so I think  this is fixed

**Comment by koniiiik** [comment]
> My dependabot started listing ignores at the top of each new PR, so I think this is fixed

As pointed out in multiple comments above, this only addresses the issue if dependabot actually opens a PR for a given dependency. If the ignore conditions prevent the creation of a PR, the only way to find the list of ignore conditions is to go digging through the debug logs of recent dependabot action runs.

**Comment by jtherrmann** [comment]
It would be great to be able to view all ignore conditions for a given repo.

**Comment by OJFord** [comment]
@greysteil @feelepxyz @asciimike @infin8x Sorry for the ping, but seems this issue should be reopened at least? It was closed as done (not won't-do) two years ago, but it isn't, or was never released.

**Comment by greysteil** [comment]
Hey @OJFord , I'm afraid none of the folks listed above work at GitHub anymore! @JamieMagee may be able to re-open the issue (which I agree shouldn't be closed as completed).

Can I humbly make a suggestion for a fix @JamieMagee?
* In an ideal world, the [dependency graph insights tab](https://github.com/dependabot/dependabot-core/network/dependencies) would be overhauled to show this kind of information. That's a major lift and I imagine isn't something you can fund
* As a stopgap, how about extending the `@dependabot show <dependency name> ignore conditions` command to be `@dependabot show [dependency name] ignore conditions`, and return a list of all the ignore conditions for a repo when the dependency name is omitted? That would give a great stopgap solution to this issue, and a natural extension of the existing command. I bet it's also a much smaller lift that's actually under the Dependabot team's control 🙏

**Comment by torokati44** [comment]
Yes, please reopen this at least.

**Comment by julioxavierr** [comment]
Come on, the streets need this

**Comment by rlconst** [comment]
Yeatrs pass and still noone adressed it

**Comment by OJFord** [comment]
@thavaahariharangit @AbhishekBhaskar @JamieMagee please see comments above - this is closed as 'done', but isn't - could you re-open? Or is it implicitly won't-do?
